### PR TITLE
[1.0] Add tests and docs for programmatically creating/modifying pages

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

File: docs/docs/creating-and-modifying-pages.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+```javascript
+// Implement the Gatsby API “onUpsertPage”. This is
+// called after every page is created.
+exports.onUpsertPage = async ({ page, boundActionCreators }) => {
+  const { upsertPage, deletePageByPath } = boundActionCreators
+
+  // Remove trailing slash
+  const oldPath = page.path
+  page.path = page.path.replace(/\/$/, "")
```
Comment (0x80) [inline docs/docs/creating-and-modifying-pages.md]: Looks like there's an upsertPage() call missing here?

Comment (KyleAMathews) [inline docs/docs/creating-and-modifying-pages.md]: Haha yes :-) Those were the sort of keen copy editing suggestions I was looking for.

Comment (0x80) [inline docs/docs/creating-and-modifying-pages.md]: Hmm maybe not. That seems to create a loop 😄

Comment (KyleAMathews) [inline docs/docs/creating-and-modifying-pages.md]: Oh right, forgot about that too... yeah probably we just need to make Gatsby not call the plugin that called `upsertPage` the first time... I'll add this in a sec.

**Pull request merged**